File: Problema3.hs
division :: (Eq a, Fractional a, Show a) => a -> a -> String
division _ 0 = "Error: división por cero"
division x y = "Resultado: " ++ show (x / y)

main :: IO ()
main = do
    putStrLn "Ingrese el primer número::"
    numInput <- getLine
    putStrLn "Ingrese el segundo número:"
    denInput <- getLine
    let numerador = read numInput :: Double
    let denominador = read denInput :: Double
    putStrLn $ division numerador denominador

File: Problema3.py
try:
    n1 = float(input("Ingrese el primer número: "))
    n2 = float(input("Ingrese el segundo número: "))
    resultado = n1 / n2
    print(f"El resultado de la división es: {resultado}")

except ZeroDivisionError:
    print("Math Error: No se puede dividir por 0.")